Feature

julianhyatt opened the pull request:
* added thirty five up order mapper
* added thirty five up persister
* added order post save expander plugin
* added order pre save expander plugin

tests will follow

Commit:
* added thirty five up order mapper
* added thirty five up persister
* added order post save expander plugin
* added order pre save expander plugin

sourcecube commented on src/FondOfSpryker/Shared/ThirtyFiveUp/Transfer/thirty_five_up.transfer.xml:
```diff
@@ -0,0 +1,35 @@
+    <transfer name="ThirtyFiveUpOrder">
+        <property name="id" type="int"/>
+        <property name="items" singular="item" type="ThirtyFiveUpOrderItem[]"/>
+        <property name="salesOrderId" type="int"/>
```
Is that the usual way of writing ? Or is it idSalesOrder..

julianhyatt replied:
Ist egal, kann ich aber ändern

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/ThirtyFiveUpConfig.php:
```diff
@@ -0,0 +1,25 @@
+    /**
+     * @return string
+     */
+    public function getAttributeSkuPrefix(): string
+    {
+        return $this->get(ThirtyFiveUpConstants::SKU_PREFIX, '_sku');
+    }
+}
```
It's a suffix in this term

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Persistence/Propel/Schema/thirty_five_up.schema.xml:
```diff
@@ -0,0 +1,40 @@
+    <table name="fos_thirty_five_up_order" phpName="ThirtyFiveUpOrder" identifierQuoting="true">
+        <column name="id_thirty_five_up_order" required="true" type="INTEGER" autoIncrement="true" primaryKey="true"/>
+
+        <column name="fk_order_id" required="false" type="INTEGER"/>
```
Does it make sense to use the name `fk_sales_order` for consistency or what is the table name releated to this FK?

julianhyatt replied:
ur right

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Business/Model/Writer/ThirtyFiveUpOrderWriter.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        return $this->entityManager->updateThirtyFiveUpOrder($orderTransfer);
+    }
+}
```
I like the idea of the writer 👍

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Business/Model/Mapper/ThirtyFiveUpOrderMapper.php:
```diff
@@ -0,0 +1,217 @@
+     *
+     * @return bool
+     */
+    protected function stringEndsWith(string $haystack, string $needle): bool
+    {
+        $length = strlen($needle);
+        if ($length === 0) {
+            return true;
+        }
+
+        return (substr($haystack, -$length) === $needle);
+    }
```
Good that this will be dropped with PHP8 :D might be worth that we create a helper package for str functions until we can switch

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Business/Model/Handler/ThirtyFiveUpOrderHandler.php:
```diff
@@ -0,0 +1,60 @@
+    {
+        $mappedData = $this->orderMapper->fromQuote($quoteTransfer);
+        if ($mappedData !== null) {
+            $quoteTransfer->setThirtyFiveUpOrder($this->orderWriter->create($mappedData));
```
What happens when `orderWriter->create ` fails ?

julianhyatt replied:
rollback for db stuff and the customer gets oops page

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Business/Model/Mapper/ThirtyFiveUpOrderMapper.php:
```diff
@@ -0,0 +1,217 @@
+
+        return $thirtyFiveUpItems;
+    }
+}
```
The mapper is quite complex in general, def needs some tests

sourcecube reviewed (commented):
No tests?

**Edit:** Just seen the description

sourcecube commented on src/FondOfSpryker/Zed/ThirtyFiveUp/Communication/Plugin/Sales/UpdateThirtyFiveUpOrderAfterPostSaveOrderExpanderPlugin.php:
```diff
@@ -0,0 +1,35 @@
+ * @method \FondOfSpryker\Zed\ThirtyFiveUp\Business\ThirtyFiveUpFacadeInterface getFacade()
+ * @method \FondOfSpryker\Zed\ThirtyFiveUp\ThirtyFiveUpConfig getConfig()
+ */
+class UpdateThirtyFiveUpOrderAfterPostSaveOrderExpanderPlugin extends AbstractPlugin implements OrderPostSavePluginInterface
```
The name doesn't feel right here, there is no "expand" function and it's super verbose.. I need to read it two to three times to barely understand what it does

Commit:
- refactoring
- added order item pre save plugin for adding vendor and vendor sku to the item

Commit:
- phpcs fixes

Commit:
- fixed naming for oms model
- added mapper for entity to transfer

Commit:
- refactoring
- added reader
- added entity mapper

Commit:
- refactoring

Commit:
- phpcs fixes

Commit:
- added store
- added unique fields
- added indices
- refactoring
